pycbc/filter/resample.py

```diff
@@ -64,24 +70,50 @@ def lfilter(coefficients, timeseries):
                           epoch=timeseries.start_time,
                           delta_t=timeseries.delta_t)
     elif (len(timeseries) < fillen * 10) or (len(timeseries) < 2**18):
+        from pycbc.strain.strain import create_memory_and_engine_for_class_based_fft
+        from pycbc.strain.strain import execute_cached_fft
+
         cseries = (Array(coefficients[::-1] * 1)).astype(timeseries.dtype)
         cseries.resize(len(timeseries))
         cseries.roll(len(timeseries) - fillen + 1)
 
         flen = len(cseries) // 2 + 1
         ftype = complex_same_precision_as(timeseries)
 
-        cfreq = zeros(flen, dtype=ftype)
-        tfreq = zeros(flen, dtype=ftype)
-
-        fft(Array(cseries), cfreq)
-        fft(Array(timeseries), tfreq)
-
-        cout = zeros(flen, ftype)
-        out = zeros(len(timeseries), dtype=timeseries)
-
-        correlate(cfreq, tfreq, cout)
-        ifft(cout, out)
+        if not USE_CACHING_FOR_LFILTER:
+            cfreq = zeros(flen, dtype=ftype)
+            tfreq = zeros(flen, dtype=ftype)
+            fft(Array(cseries), cfreq)
+            fft(Array(timeseries), tfreq)
+            cout = zeros(flen, ftype)
+            correlate(cfreq, tfreq, cout)
+            out = zeros(len(timeseries), dtype=timeseries)
+            ifft(cout, out)
+
+        else:
+            npoints = len(cseries)
+            # NOTE: This function is cached!
+            ifftouts = create_memory_and_engine_for_class_based_fft(
+                npoints,
+                timeseries.dtype,
+                ifft=True,
+                uid=486876761
+            )
```

Comment:
I'm assuming that the number you select for the uid here is just so that it's different from anything else? If there's a reason it has this value (also below) then please add a code comment explaining what determines the value.

If it's just to be unique from other calls of the same function, would it be clearer to just define some constants in this file, like `LFILTER_UNIQUE_ID = 1`, or whatever?

Reply:
Thanks Josh. This is just to give different outputs if called with the same inputs from different functions. The number just has to be unique. I made this a constant, with some inline comments, as suggested.